### DISCUSSION TITLE
How can I force videos to index at local time instead of UTC time?

### DISCUSSION THREAD

**Discussion by jrstrunk** [photoprism/photoprism#4154] [Q&A]
All my .mp4 video imports show in my library as UTC time instead of local time. Is there a way I can import them so that the local time / offset are saved? Do I need to create a sidecar file in a specific format? I have tried creating .json and .xml sidecar files with a "CreationDate" field like "2023:06:02 12:01:51+02:00", but it does not work for videos - though using the photoshop:CreationDate tag for other file types (like gifs) seems to work.

Edit: actually now I see not all of them get UTC time. I have a nightly backup and index from my phone, and it looks like if the video was taken alongside photos then it gets the correct time zone? I am not quite sure what is making the difference, but it would be nice to be able to have all videos be in local time. I do not mind setting up import middleware if I need to generate a sidecar file first for reliability.

Edit: After some more testing, it seems that any lone video (with no photos taken alongside it) will be given a UTC date when it is taken in a local time that is in a different calendar date when converted to UTC time. All other videos imported have the local time. I noticed also when manually updating a video's time metadata, if you update the timezone the time will change, but the date will stay the same.

It would be nice to be able to provide a partial sidecar file when importing to override only certain metadata properties.

**Comment by Jimmy-eng221** [comment]
Forcing videos to index at local time instead of UTC can be tricky, and it seems you've done some thorough testing to understand the behavior. Here's what I've found:

- Your observation is correct. Lone videos often get assigned UTC time, especially if the capture time in your local time zone translates to a different calendar date in UTC.

- This inconsistency can be frustrating, especially when managing a large media library.

**Potential Solutions:**

Batch Editing with Software:

Some media management software allows batch editing of video metadata. You can potentially set the time zone and date information for all videos at once. However, this functionality varies depending on the specific software.

Sidecar Files (Limited Use):

While your experience suggests limited success with sidecar files for videos, it might be worth exploring software that specifically supports sidecar files with video imports. This approach could work, but compatibility might be limited.

Time Management Tools for Consistency:

While not a direct solution for forcing local time indexing, consider utilizing time management tools during video capture. There are many tools for recording when capturing videos, especially if you're working across different time zones. Also [Date and time calculator ](https://datetimecheck.com/)also offers features like time zone converters and scheduling tools, which can be helpful for maintaining accurate timekeeping during the video capture process.

Software Updates and Community Forums:

Keep an eye out for updates to your media management software. Developers might address this specific issue in future versions. Additionally, check the software's user forums or online communities. Other users might have encountered similar challenges and found workarounds.

**Reply by lastzero** [reply]
Note that the UTC time zone default for certain video types is not an accidental inconsistency, but part of the official standard for these formats. See the public source code for details.
